### PR TITLE
Dinic Algorithm

### PR DESCRIPTION
**Pull request by kapil4457**
## 🛠️ Issue (Number)

Issue no #486 

## ✔️ Check List (Check all the applicable boxes) 

- [✔️] My code follows the code style of this project.
- [✔️] This PR does not contain plagiarized content.
- [✔️] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

I have added Dinic's Algorithm in java , c++ and python

## 📷 Screenshots
Capture

### CODE REVIEW

**Review by Arun9739 — CHANGES_REQUESTED** [review]
Code is Super!
@kapil4457 But take input from user, instead of giving it directly in a program.
Also codefactor is showing one error, plz resolve it.

**Comment by kapil4457** [comment]
> Code is Super! @kapil4457 But take input from user, instead of giving it directly in a program. Also codefactor is showing one error, plz resolve it.
Please check now.. I have made the changes you asked for

**Review by 072arushi — APPROVED** [review]

**Review by Rishikavishnoi — APPROVED** [review]

File: C++/Graphs/Dinic.cpp [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    cin >> m;
+
+    cout << "Enter the input for each edge in the form u , v, C where u = starting node , v = end node and C = capacity : " << endl;
+    int u, v, c;
+    for (int i = 0; i < n; i++)
+    {
+        cin >> u, v, c;
+        g.addEdge(u, v, c);
+    }
+    // g.addEdge(0, 1, 16);
```
Comment (Kumar-laxmi) [inline C++/Graphs/Dinic.cpp:191]: Why have you commented these lines

Comment (kapil4457) [reply C++/Graphs/Dinic.cpp:191]: > Why have you commented these lines

I have given sample graph examples to the user for reference.

**Review by Rancho2002 — CHANGES_REQUESTED** [review]
Uncomment some addGraph() portion and also give code snippet of user input

**Review by Rancho2002 — APPROVED** [review]

**Pull request merged**